### ISSUE TITLE
Error typescript with the change on #128

### ISSUE THREAD

**Issue by Julienng** [hoangvvo/next-connect#132]
Hi,

I just upgrade the next-connect on 0.10.1 and the change :

```
function NextConnect<U,V>(req: U, res: V): Promise<void>;
```

into the NextConnect interface make typescript infer any on req & res on `.use` function.

Example screenshot before / after

<img width="828" alt="before_128" src="https://user-images.githubusercontent.com/8990614/116543078-b5752180-a8ed-11eb-97d7-2536de55331b.png">

<img width="775" alt="after_128" src="https://user-images.githubusercontent.com/8990614/116543092-b9a13f00-a8ed-11eb-879e-375b52bb5dfa.png">

I don't know enough about TS inference here to make a pull request 😕.

**Comment by hoangvvo** [comment]
![image](https://user-images.githubusercontent.com/40987398/116819119-c7c6b800-ab98-11eb-99b9-e8de4e3279b2.png)

That is very strange. I used it and personally had no problem. However though, I did notice this happens from time to time (even on older version) but not sure the cause.

**Comment by Julienng** [comment]
Yes, this is very strange 🤔. Ts is weird sometimes! Thanks anyway, your lib is awesome 👍!
